src/screens/FeedScreen/hooks/useCoreFeed.js

```diff
@@ -60,56 +60,10 @@ const useCoreFeed = () => {
     }
   };
 
-  const handleBgContentFeed = (feed) => {
-    if (feed.anon_user_info_color_code) {
-      const rgb = hexToRgb(feed?.anon_user_info_color_code, 0.25);
-      const color = {
-        bg: `${rgb}`,
-        color: 'rgba(0,0,0)'
-      };
-      return color;
-    }
-    const randomIndex = getRandomInt(0, listFeedColor.length);
-    let newColor = listFeedColor[randomIndex];
-    newColor = {
-      ...newColor,
-      bg: hexToRgb(newColor.bg, 0.25),
-      color: 'rgba(0,0,0)'
-    };
-    return newColor;
-  };
-
-  const mappingColorFeed = ({dataFeed, dataCache}) => {
-    if (dataCache && typeof dataCache === 'string') {
-      dataCache = JSON.parse(dataCache);
-    }
-    const mapNewData = dataFeed?.map((feed) => {
-      const cacheBg = dataCache?.find((cache) => cache?.id === feed?.id);
-      if (cacheBg?.bg) {
-        const isHexColor = checkIsHasColor(cacheBg.bg);
-        if (isHexColor) {
-          return {...feed, bg: hexToRgb(cacheBg.bg, 0.25)};
-        }
-        return {...feed, bg: cacheBg?.bg};
-      }
-      return {...feed, ...handleBgContentFeed(feed)};
-    });
-
-    return {
-      mapNewData
-    };
-  };
-
   const handleDataFeeds = (dataFeeds, offsetFeed = 0) => {
     if (dataFeeds.data.length > 0) {
       const {data} = dataFeeds;
-      let stringCacheFeed = StorageUtils.feedPages.get();
-      if (stringCacheFeed) {
-        stringCacheFeed = JSON.parse(stringCacheFeed);
-      } else {
-        stringCacheFeed = {};
-      }
-      const {mapNewData} = mappingColorFeed({dataFeed: data, dataCache: stringCacheFeed?.data});
+      const mapNewData = mappingColorFeed(data);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change simplifies the handling of data feeds by removing the need to check for cached data and handle background colors directly within this function. Instead, it delegates these responsibilities to the `mappingColorFeed` function from the `useMappingPostColorHook`. This improves the separation of concerns and makes the code easier to understand and maintain.
```diff
- const handleBgContentFeed = (feed) => {...};
- const mappingColorFeed = ({dataFeed, dataCache}) => {...};
- let stringCacheFeed = StorageUtils.feedPages.get();
- if (stringCacheFeed) {
-   stringCacheFeed = JSON.parse(stringCacheFeed);
- } else {
-   stringCacheFeed = {};
- }
- const {mapNewData} = mappingColorFeed({dataFeed: data, dataCache: stringCacheFeed?.data});
+ const mapNewData = mappingColorFeed(data);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->